### PR TITLE
chore(CI): Switch regression detector to new API and analysis service

### PR DESCRIPTION
**Pull request by GeorgeHahn**
This should be a transparent change. If you notice any issues, please ping us in #single-machine-performance. We'll also be keeping a close eye on things after this is merged.

## Single Machine Performance API (`curta`)

The Single Machine Performance team has built an API to interact with the regression detector system. This will allow us to maintain reliability while continuing to evolve the system. This is enabled with the `--use-curta` flag, which will become the default in an upcoming release.

The curta API uses a unique URI for each team. This isn't secret - all endpoints have authorization - but we'd prefer it not be public. The URI for Vector is in SMP-617. The `smp` binary can read this from the `SINGLE_MACHINE_PERFORMANCE_API` environment variable or the `--api-base` flag.

## Lambda Analysis Worker (`consignor`)

We have also changed how the analysis worker is deployed so it can run on-demand for incoming requests. This improves analysis speed while reducing our operational workload. This is set using the `--use-consignor` flag, which will also become the default in a future release.

### CODE REVIEW

File: .github/workflows/regression.yml [hunk 1]
```diff
@@ -456,6 +456,8 @@ jobs:
       - compute-metadata
       - upload-baseline-image-to-ecr
       - upload-comparison-image-to-ecr
+    env:
+      SINGLE_MACHINE_PERFORMANCE_API: ${{ secrets.SINGLE_MACHINE_PERFORMANCE_API }}
```
Comment (neuronull) [inline .github/workflows/regression.yml:460]: As this is a protected workflow, any reason not to set this once at the top instead of in multiple jobs?

Comment (GeorgeHahn) [reply .github/workflows/regression.yml:460]: I'll double check - my reading was that github doesn't support setting this above the `job` level.

Comment (neuronull) [reply .github/workflows/regression.yml:460]: Here is an example where we do that for different env vars: https://github.com/vectordotdev/vector/blob/251c4c4608a70fd6c112ecacd0517c301f21e33c/.github/workflows/test.yml#L13-L28

Comment (GeorgeHahn) [reply .github/workflows/regression.yml:460]: You're completely correct. I'll clean this up. Thanks, this will be a nice cleanup in the SMP repo as well.

Comment (GeorgeHahn) [reply .github/workflows/regression.yml]: Done in [6e4f7c3](https://github.com/vectordotdev/vector/pull/17912/commits/6e4f7c3ff1c42cf2d8c74d73bb68838349005afd)

**Comment by neuronull** [comment]
It looks like the integration test failure is unrelated. We may have a problem on the master branch.

But for the spell checking error, unfortunately you'll probably have to add an exception to `.github/actions/spelling/allow.txt`

**Review by spencergilbert — APPROVED** [review]

**Review by jszwedko — APPROVED** [review]

**Comment by GeorgeHahn** [comment]
It looks like I don't have permission to merge this, would y'all mind doing that? Thank you!

**Pull request merged**